components/path/path.js

```diff
@@ -50,24 +50,24 @@ PathViewModel.prototype.updateAnimationFrame = function(deltaT) {
 }
 PathViewModel.prototype.updateStatus = function() {
   var self = this;
-  this.server.get('/quickstatus', { path: this.repoPath() }, function(err, status){
-    self.loadingProgressBar.stop();
-    if (err) return;
-    if (status.type == 'inited' || status.type == 'bare') {
-      if (self.repoPath() !== status.gitRootPath) {
-        self.repoPath(status.gitRootPath);
-        programEvents.dispatch({ event: 'navigated-to-path', path: self.repoPath() });
-        programEvents.dispatch({ event: 'working-tree-changed' });
-      }
-      self.status(status.type);
-      if (!self.repository()) {
-        self.repository(components.create('repository', { server: self.server, path: self }));
+  return this.server.getPromise('/quickstatus', { path: this.repoPath() })
```

Comment:
Because right now, as it stands we get several warning such as below.  This was my attempt at possibly removing this warning but I'm still getting these and I plan to revisit this issue once promisifying is finished.

```
Warning: a promise was created in a handler at /127.0.0.1:8333/plugins/remotes/remotes.bundle.js:78:16 but was not returned from it, see http://goo.gl/rRqMUw
    at new Promise (http://127.0.0.1:8333/js/ungit.js:19300:14)
    at Server.queryPromise (http://127.0.0.1:8333/js/ungit.js:7683:10)
    at Server.getPromise (http://127.0.0.1:8333/js/ungit.js:7719:15)
    at RemotesViewModel.fetch (http://127.0.0.1:8333/plugins/remotes/remotes.bundle.js:50:47)
    at http://127.0.0.1:8333/plugins/remotes/remotes.bundle.js:78:16
From previous event:
    at RemotesViewModel.updateRemotes (http://127.0.0.1:8333/plugins/remotes/remotes.bundle.js:64:6)
    at new RemotesViewModel (http://127.0.0.1:8333/plugins/remotes/remotes.bundle.js:34:8)
    at http://127.0.0.1:8333/plugins/remotes/remotes.bundle.js:10:10
    at Object.components.create (http://127.0.0.1:8333/js/ungit.js:52258:10)
    at new RepositoryViewModel (http://127.0.0.1:8333/plugins/repository/repository.bundle.js:20:29)
    at http://127.0.0.1:8333/plugins/repository/repository.bundle.js:9:10
    at Object.components.create (http://127.0.0.1:8333/js/ungit.js:52258:10)
    at http://127.0.0.1:8333/plugins/path/path.bundle.js:64:38
From previous event:
    at PathViewModel.updateStatus (http://127.0.0.1:8333/plugins/path/path.bundle.js:55:6)
    at PathViewModel.shown (http://127.0.0.1:8333/plugins/path/path.bundle.js:46:8)
    at init (http://127.0.0.1:8333/js/ungit.js:7499:20)
    at http://127.0.0.1:8333/js/ungit.js:27413:46
    at Object.ignore (http://127.0.0.1:8333/js/ungit.js:25571:33)
    at http://127.0.0.1:8333/js/ungit.js:27412:48
    at Object.arrayForEach (http://127.0.0.1:8333/js/ungit.js:24269:17)
    at applyBindingsToNodeInternal (http://127.0.0.1:8333/js/ungit.js:27398:22)
    at applyBindingsToNodeAndDescendantsInternal (http://127.0.0.1:8333/js/ungit.js:27278:37)
    at Object.ko.applyBindings (http://127.0.0.1:8333/js/ungit.js:27489:9)
    at http://127.0.0.1:8333/js/ungit.js:29347:24
    at invokeForEachNodeInContinuousRange (http://127.0.0.1:8333/js/ungit.js:29298:13)
    at activateBindingsOnContinuousNodeArray (http://127.0.0.1:8333/js/ungit.js:29345:13)
    at executeTemplate (http://127.0.0.1:8333/js/ungit.js:29393:13)
    at ko.dependentObservable.disposeWhen (http://127.0.0.1:8333/js/ungit.js:29435:46)
    at Function.evaluateImmediate_CallReadThenEndDependencyDetection (http://127.0.0.1:8333/js/ungit.js:26252:103)
    at Function.evaluateImmediate_CallReadWithDependencyDetection (http://127.0.0.1:8333/js/ungit.js:26224:29)
    at Function.evaluateImmediate (http://127.0.0.1:8333/js/ungit.js:26188:18)
    at Object.ko.computed.ko.dependentObservable (http://127.0.0.1:8333/js/ungit.js:26049:28)
```